File: backend/app/schemas/audit.py
from pydantic import BaseModel
from typing import Optional, Any
from datetime import datetime

class AuditLogOut(BaseModel):
    id: int
    user_id: int
    action_type: str
    resource_id: str
    old_values: Optional[Any] = None
    new_values: Optional[Any] = None
    ip_address: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/audit.ts
export interface AuditLogOut {
  id: number;
  user_id: number;
  action_type: string;
  resource_id: string;
  old_values: any | null;
  new_values: any | null;
  ip_address: string | null;
  created_at: string;
}
